homeassistant/components/switch/config_flow.py

```diff
@@ -0,0 +1,33 @@
+"""Config flow for Switch integration."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.helpers import selector
+
+from .const import DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("entity_id"): selector.selector({"entity": {"domain": "switch"}}),
+        vol.Required("name"): selector.selector({"text": {}}),
+    }
+)
+
+
+class SwitchLightConfigFlowHandler(
+    config_entries.HelperConfigFlowHandler, domain=DOMAIN
+):
+    """Handle a config or options flow for Switch Light."""
+
+    _schema = DATA_SCHEMA
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
```

Comment:
I think that we can create this method automatically in `__init_subclass__`

Reply:
you can even pass the schema to `__init_subclass__` but not sure if that's better/worse

Reply:
That's great, refactored to do like that.